### PR TITLE
Don't drop error info in ErrorBoundary

### PR DESCRIPTION
**Pull request by akalin-keybase**
Log a plain object instead of the Error object directly.

Display/log the error stack trace.

Display newlines, too.

Add better typing.

### CODE REVIEW

File: shared/common-adapters/error-boundary.js [hunk 1]
```diff
@@ -42,43 +61,52 @@ const Fallback = ({error, info}: *) => {
...
-  componentWillReceiveProps(nextProps: any) {
+class ErrorBoundary extends React.PureComponent<Props, State> {
+  state = {info: null}
+
+  componentWillReceiveProps(nextProps: Props) {
     if (this.props.children !== nextProps.children) {
-      if (this.state.error) {
-        this.setState({error: null, info: null})
-      }
+      this.setState({})
```
Comment (chrisnojima) [inline shared/common-adapters/error-boundary.js]: I don't think you want this. This means anything that gets passed to this might clear it out

Comment (akalin-keybase) [reply shared/common-adapters/error-boundary.js]: Hmm...how is the old behavior different, though? What should I do instead?

Comment (chrisnojima) [reply shared/common-adapters/error-boundary.js]: Oh sorry. No I mean setstate with empty does nothing. No update

Comment (akalin-keybase) [reply shared/common-adapters/error-boundary.js]: Ah...it's a merge. Got it.

**Pull request merged**